### ISSUE TITLE
extractWithoutRootDirectory() fails if destination is .

### ISSUE THREAD

**Issue by xabbuh** [raulfraile/distill#22]
If you pass `.` (the current working directory) as the destination to the `Distill::extractWithoutRootDirectory()` method, extracting the archive fails with a message like this:

```
rmdir: failed to remove ‘.’: Invalid argument
```

The reason for this is, for example, explained [here](http://superuser.com/questions/708804/removing-while-standing-on-it?answertab=active#tab-top).

**Comment by raulfraile** [comment]
@xabbuh thanks for opening the issue! Working on it...

**Comment by xabbuh** [comment]
@raulfraile Thank you. I thought that calling `realpath()` on the path that is passed to `rmdir()` would suffice. However, that led to other issues (probably due to the fact that the current working directory does not exist any more after the removal). Maybe we need to switch the working directory first in those cases.

**Comment by raulfraile** [comment]
@xabbuh yes, I'm working with that approach, the working directory needs to be changed first

**Comment by raulfraile** [comment]
@xabbuh it should be fixed in the `fix/current_dir` branch. Do you have any comments about https://github.com/raulfraile/distill/commit/eb8fed8e356923d212c0aef2da02e6272ebe78dd?

**Comment by xabbuh** [comment]
@raulfraile Do we need some special handling in case there is no parent directory?

**Comment by raulfraile** [comment]
@xabbuh would that happen only if you are in the root directory or is there any other situation? Anyway, I think we should handle gracefully and throw an error

**Comment by xabbuh** [comment]
I think this will only happen when you are in the root directory.

**Comment by xabbuh** [comment]
@raulfraile This fix does not work as is. I think the issue is the check `if ('.' === $path)`. Imho this needs to be either `if ('.' === basename($path)` or something like `if (getcwd() === realpath($path))`.

**Comment by raulfraile** [comment]
@xabbuh thanks for your suggestions! Now I throw an exception when you are in the root directory and I use `getcwd() === realpath($path)` to check if the target directory is the same as the current directory

**Comment by raulfraile** [comment]
Commit: https://github.com/raulfraile/distill/commit/5fa7ce1d837953eb26247a4160586c023ae88023

**Comment by xabbuh** [comment]
@raulfraile I can confirm that this fixes the root issue.

**Comment by xabbuh** [comment]
@raulfraile I see one issue with the proposed solution: When using the Symfony installer, the user would have to change directories after installing Symfony in the current directory because the current working directory does not exist anymore. What do you think about extracting the archive to a temporary subdirectory instead and moving contents up after the extraction succeeded?
